lib/elixir/lib/integer.ex

```diff
@@ -400,17 +400,23 @@ defmodule Integer do
       iex> Integer.gcd(10, 0)
       10
 
+      iex> Integer.gcd(7, 7)
+      7
+
       iex> Integer.gcd(0, 0)
       0
 
   """
+  @spec gcd(0, 0) :: 0
+  @spec gcd(int, int) :: int when int: pos_integer
```

Comment:
@michalmuskala the doc clarifies later:

> The specification with the X type variable specifies that the function takes a tuple and returns the same tuple.

In that context "same" means that tuple should be of the same shape and contain same type on the same positions, not exactly the same term.